libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/RepeatDialog.cs

```diff
@@ -58,10 +70,16 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
             var repeatedIds = dc.State.GetValue<List<string>>(TurnPath.RepeatedIds, () => new List<string>());
             if (repeatedIds.Contains(targetDialogId))
             {
-                throw new ArgumentException($"Recursive loop detected, {targetDialogId} cannot be repeated twice in one turn.");
+                if (this.AllowLoop == null || this.AllowLoop.GetValue(dc.State) == false)
+                {
```

Comment:
I would move this conidtion around the entire repeatedIds section, no need to track ids if you aren't using them.